### ISSUE TITLE
Docs: "how to extend" checklist

### ISSUE THREAD

**Issue by wardbell** [johnpapa/angular-ngrx#68]
List of docs need for ngrx-data extension points and techniques:

- [ ] Add a new entity type
- [ ] Use an entity data service directly without going through store
- [ ] Add a custom entity data service that doesn't inherit from BasicDataService
- [ ] Add a pluralization term (e.g., {box: boxes}
- [ ] Replace the pluralizer entirely
- [ ] Add a custom entity selectors 
- [ ] Add a custom entity dispatcher
- [ ] extend/override the entity reducer
- [ ] extend/override the entity effects
- [ ] add a reducer that does something extra with one of the EntityActions
- [ ] Block add/update/delete for an entity type
- [ ] Clear the entity cache
- [ ] Add a completely different feature to the store, unrelated to entities

**Comment by johnpapa** [comment]
For our notes later
![image](https://user-images.githubusercontent.com/1202528/33792664-3233dcec-dc74-11e7-88d7-bf08207e2f5b.png)

**Comment by wardbell** [comment]
Will make the pk (`id`) configurable before too long as I’ve got `entity.id` in too many places and it will just get worse.  I will create a client-side id-generator option while I’m at it.

Also fixing the filter thing today. My plan:
* A filter is an object:
```filter = {
  name: string, // name of a filter function in the `EntityFiltersService`
  value: any, // typically a string but can be anything as long as it is serializable
}
```
* a request for filtering provides at least the value. If the name is unspecified, logic uses the “default” filter.
* Will have default implementation of `EntityFilterService` that has a `{"default": today's-name`-filter}` so no changed behavior.
* Developer can register new filters or replace the service itself

**NGRX-DATA CODING RULES/STANDARDS**
I’m following certain redux-ie rules which will be documented:
* Action instances are immutable
* Action instances are serializable (because the tools demand it and we should be able to persist them)
* All store values are immutable and serializable
* All reducers and effects are “pure” functions (I think I’ve held true to this)
* An Action whose name begins “_” is an _internal action_, created by effects, and should not be dispatched or selected by application code (although nothing can stop you).  An example is `"_DELETE"` which is the one that actually calls the data service. 

I’m sure I have other guidelines but I’m trying hard to follow these. The “serializable” rule is the main reason I got rid of `EntityAction.entityType` which was a class (i.e, a function).

**Comment by wardbell** [comment]
Look at closed PR #84 for documentation ideas about EntityFilters.
